clients/feeder/feeder.go

```diff
@@ -262,14 +273,23 @@ func (c *Client) get(ctx context.Context, queryURL string) (io.ReadCloser, error
 				res.Body.Close()
 			}
 
+			c.curTimeout++
+			if c.curTimeout >= len(c.timeouts) {
+				c.curTimeout = len(c.timeouts) - 1
+			}
+
```

Comment:
This functionality should be part of the `Timeouts` type

Reply:
moved